snippets/scripts/jira-components-to-compass-components/migrateJiraComponentsToCompassComponents.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python3
+
+import getpass
+import time
+
+import requests
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import HTTPError
+import urllib.parse
+
+DEFAULT_COMPASS_COMPONENT_TYPE = "SERVICE"
+
+COMPASS_CREATE_COMPONENT_MUTATION = """
+mutation createComponent(,$cloudId:ID!, $input: CreateCompassComponentInput!) {
+  compass {
+    createComponent(cloudId: $cloudId, input:$input) {
+      success
+      componentDetails {
+        id
+      }
+    }
+  }
+}
+"""
+
+TENANT_INFO_QUERY = """
+query tenantInfo($hostNames: [String!]) {
+  tenantContexts(hostNames:$hostNames) {
+    cloudId
+  }
+}
+"""
+
+
+def make_api_call(path, method, data=None, headers=None, no_response=False):
+    # print(f"Making a {method} request to {url}")
+    try:
+        response = requests.request(
+            method,
+            f"https://{DOMAIN_NAME}{path}",
+            auth=HTTPBasicAuth(USER_NAME, API_TOKEN),
+            json=data,
+            headers=headers
+        )
+        response.raise_for_status()
+        if no_response:
+            return
+        return response.json()
+    except HTTPError as http_err:
+        if http_err.response.status_code == 404:
+            return None
+        print(f"HTTP error occurred: {http_err} - {http_err.response.text}")
+        # rethrow the exception
+        raise http_err
+    except Exception as err:
+        print(f"An error occurred: {err}")
+
+
+def make_gql_api_call(query, variables):
+    url = f"/gateway/api/graphql"
+    headers = {
+        'Content-Type': 'application/json'
+    }
+    data = {
+        "query": query,
+        "variables": variables
+    }
+    return make_api_call(url, "POST", data, headers)
+
+
+def get_jira_component_list(project_key):
+    url = f"/rest/api/3/project/{project_key}/components"
```

Comment:
This endpoint is paginated, right? 👀

Reply:
No this is the one that returns ALL components!